tools/deb/istio-iptables.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Initialization script responsible for setting up port forwarding for Istio sidecar.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+  echo "${0} -p ENVOY_PORT -u UID -s IN_SERVICE_PORTS [-h]"
+  echo ''
+  echo '  -p: Specify the envoy port to which redirect all TCP traffic'
+  echo '  -u: Comma separate list of UIDs for which the redirection is not'
+  echo '      applied. Typically, this is the UID of the proxy container'
+  echo '  -i: Comma separated list of IP ranges in CIDR form to redirect to envoy (optional)'
+  echo '  -s: Comma separated list of incoming service IPs to intercept (optional)'
+  echo ''
+}
+
+IP_RANGES_INCLUDE=""
+
+while getopts ":p:u:e:i:h:s" opt; do
+  case ${opt} in
+    p)
+      ENVOY_PORT=${OPTARG}
+      ;;
+    u)
+      ENVOY_UID=${OPTARG}
+      ;;
+    i)
+      IP_RANGES_INCLUDE=${OPTARG}
+      ;;
+    s)
+      ISTIO_LOCAL_PORTS=${OPTARG}
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${ENVOY_PORT-}" ]]; then
+  ENVOY_PORT=15001
+fi
+
+if [[ -z "${ENVOY_UID-}" ]]; then
+  ENVOY_UID=`id -u ${ENVOY_USER-istio}`
```

Comment:
nit : ` is discouraged nowadays, use $(id ...)

more important: so this will fail if the user is missing ? (which is good)
maybe explicitly check for ENVOY_UID not found with an explicit error message ?